src/plugins/intel_cpu/src/cpu_streams_calculation.cpp

```diff
@@ -17,6 +17,38 @@ using namespace InferenceEngine;
 namespace ov {
 namespace intel_cpu {
 
+std::vector<std::vector<int>> apply_core_type(const ov::hint::SchedulingCoreType input_type,
+                                              const std::vector<std::vector<int>> proc_type_table) {
+    std::vector<std::vector<int>> result_table = proc_type_table;
+
+    switch (input_type) {
```

Comment:
From my understanding the logic below is very generic and shouldn't be part of particular plugin. That also makes sense as long as we will need the same logic for GPU. So my proposal is to have common function inside dev API.

Reply:
updated